python/dalex/dalex/_arena/object.py

```diff
@@ -0,0 +1,670 @@
+        if not self.server_thread:
+            raise Exception('Server is not running')
+        self._stop_server()
+        self.server_thread.join()
+        self.server_thread = None
+
+    def update_timestamp(self):
+        """Updates timestamp
+
+        Notes
+        -------
+        This function must be called from mutex context
+        """
+        now = datetime.now()
+        self.timestamp = datetime.timestamp(now)
+
+    def clear_cache(self, plot_type=None):
+        """Clears cache
+        
+        Parameters
+        -----------
+        plot_type : str or None
+            If None all cache is cleared. Otherwise only plots with
+            provided plot_type are removed.
+
+        Notes
+        -------
+        This function must be called from mutex context
+        """
+        if plot_type is None:
+            self.cache = []
+        else:
+            self.cache = list(filter(lambda p: p.plot_type != plot_type, self.cache))
+        self.update_timestamp()
+
+    def find_in_cache(self, plot_type, params):
+        """Function searches for cached plot
+
+        Parameters
+        -----------
+        plot_type : str
+            Value of plot_type field, that requested plot must have
+        params : dict
+            Keys of this dict are params types (model, observation, variable, dataset)
+            and values are corresponding params labels. Requested plot must have equal
+            params field.
+
+        Returns
+        --------
+        PlotContainer or None
+        """
+
+        _filter = lambda p: p.plot_type == plot_type and params == p.params
+        with self.mutex:
+            result = next(filter(_filter, self.cache), None)
+        return result
+    
+    def put_to_cache(self, plot_container):
+        """Puts new plot to cache
+
+        Parameters
+        -----------
+        plot_container : PlotContainer
+        """
+        if not isinstance(plot_container, PlotContainer):
+            raise Exception('Invalid plot container')
+        with self.mutex:
+            self.cache.append(plot_container)
+
+    def fill_cache(self, fixed_params={}):
+        """Generates all available plots and cache them
+
+        This function tries to generate all plots that are not cached already and
+        put them to cache. Range of generated plots can be narrow using `fixed_params`
+
+        Parameters
+        -----------
+        fixed_params : dict
+            This dict specifies which plots should be generated. Only those with
+            all keys from `fixed_params` present and having the same value will be
+            calculated.
+        """
+        if not isinstance(fixed_params, dict):
+            raise Exception('Params argument must be a dict')
+        for plot_class in self.get_supported_plots():
+            required_params = plot_class.info.get('requiredParams')
+            if [k for k in fixed_params.keys() if not k in required_params]:
+                continue
```

Comment:
This looks complicated, or at least unintuitive to me. 
1. `if []` acts like `False` while `[] == False` acts like `False` (btw. `if [None]` acts like `True` while `[None] == True` acts like `False`)
2. `if ...: continue` can be just changed to `if not ...: do`